src/Compilers/Core/Portable/DiagnosticAnalyzer/DiagnosticAnalysisContextHelpers.cs

```diff
@@ -36,6 +43,14 @@ internal static void VerifyArguments(Diagnostic diagnostic, Func<Diagnostic, boo
             {
                 throw new ArgumentException(string.Format(CodeAnalysisResources.UnsupportedDiagnosticReported, diagnostic.Id), nameof(diagnostic));
             }
+
+            if (string.IsNullOrEmpty(diagnostic.Id) || !diagnostic.Id.All(UnicodeCharacterUtilities.IsIdentifierPartCharacter))
```

Comment:
However, for performance reasons you should redirect current clients of `SyntaxFacts` to call the new API directly to avoid additional overhead during scanning of C# source code.